flume-ng-sinks/flume-hdfs-sink/src/test/java/org/apache/flume/sink/hdfs/TestHDFSEventSink.java

```diff
@@ -1323,6 +1323,64 @@ public void testCloseOnIdle() throws IOException, EventDeliveryException, Interr
     fs.close();
   }
 
+  @Test
+  public void testBlockCompressSequenceFileWriterSync() throws IOException, EventDeliveryException, InterruptedException {
+    String hdfsPath = testPath + "/sequenceFileWriterSync";
+    String body = "test event-" + System.nanoTime();
+
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+
+    // Since we are reading a partial file we don't want to use checksums
+    fs.setVerifyChecksum(false);
+    fs.setWriteChecksum(false);
+
+    Path dirPath = new Path(hdfsPath);
+    fs.delete(dirPath, true);
+    fs.mkdirs(dirPath);
+
+    Context context = new Context();
+    context.put("hdfs.path", hdfsPath);
+    // Ensure the file isn't closed and rolled, we are only going to send one event
+    context.put("hdfs.rollCount", "10");
+    context.put("hdfs.rollSize", "0");
+    context.put("hdfs.rollInterval", "0");
+    context.put("hdfs.batchSize", "1");
+    context.put("hdfs.fileType", "SequenceFile");
+    context.put("hdfs.codeC", "BZip2Codec"); // Compression codec that doesn't require native hadoop libraries
+    context.put("hdfs.writeFormat", "Writable");
+    Configurables.configure(sink, context);
+
+    Channel channel = new MemoryChannel();
+    Configurables.configure(channel, context);
+
+    sink.setChannel(channel);
+    sink.start();
+
+    Transaction txn = channel.getTransaction();
+    txn.begin();
+    Event event = new SimpleEvent();
+    event.setBody(body.getBytes());
+    channel.put(event);
```

Comment:
> Add multiple events to the channel.
> Please increase your batch size to at least 2 events

The idea behind this test (and this is why your request for documentation is a good one ;)) is to simulate what happens when a batch of events is written to the file (and thus hsync'd to hdfs) but the file is _not_ yet closed.  In that case, the data that we wrote should still be readable.  With that in mind, I thought that writing one event (and thus a corresponding batch size of one) would suffice.  Is there a scenario that I'm missing where multiple events would be different?

Thanks!